lib/wasix/src/os/console/mod.rs

```diff
@@ -205,7 +205,34 @@ impl Console {
             }
         };
 
-        let wasi_opts = webc::metadata::annotations::Wasi::new(prog);
+        let command = if let Some(entrypoint) = pkg.entrypoint_cmd.as_deref() {
+            pkg.get_command(entrypoint).unwrap()
+        } else {
+            match pkg.commands.as_slice() {
+                [] => {
+                    return Err(SpawnError::Other(
+                        anyhow::anyhow!("The WEBC file doesn't contain any executable commands")
+                            .into(),
+                    ))
+                }
+                [one] => one,
+                [first, ..] => first,
+            }
+        };
+
+        let mut wasi_opts = command
+            .metadata()
+            .wasi()
+            .ok()
+            .and_then(|x| x)
+            .unwrap_or_else(|| webc::metadata::annotations::Wasi::new(prog));
+
+        // Prevent adding default args if custom args are specified.
+        if !args.is_empty() {
+            if let Some(wargs) = &mut wasi_opts.main_args {
+                wargs.clear();
+            }
+        }
```

Comment:
FYI - This doesn't line up with what the `wasmer` CLI does.

The `wasmer` CLI will start with the arguments provided in `[command.annotations.wasi]` and **append** the arguments passed in via the command-line.

@theduke I'll let you decide whether you are okay with having different behaviour - just thought I'd point out the possible inconsistency.

Reply:
Yeah we briefly talked about this already.
We need some solution here IMO, because often (usually) you'll want to override the args, not amend them.